Swgs

Istaisa opened the pull request:
SWGS app still in development, changes needed for showing initial data to clinical scientists.

Changes in this pull request:

- Scripts to upload/update panels from either the panelapp API or a CSV list of genes
- Gene lists for initial somatic panels
- Set up for indications (lots of many to many fields so we can add multiple panels to a tier)
- Updated tiering display so now there's 4 tables: Tier 1 somatic, Tier 2 somatic, Tier 1 germline, Tier 3 germline - closes #132 
- We can talk about this one but off the back of Joseph's artefact checking training there were some questions in the chat about MNVs and things being missed which I've realised we could also be at risk of, so I've also displayed non-intronic possible MNVs in their respective tiers - the lists are already really short and it's only added one to my test data but we can tweak this later on

Still to come!
- Tests
- Refactoring basically everything
- Documentation

erikwaskiewicz commented on swgs/management/commands/update_panels_from_csv.py:
```diff
@@ -0,0 +1,155 @@
+                old_panel_genes = []
+
+            else:
+                panel_version = previous_panel.panel_version
```
Should this be +1?

Istaisa replied:
yes. changed

erikwaskiewicz commented on swgs/management/commands/update_panels_from_csv.py:
```diff
@@ -0,0 +1,155 @@
+    """
+    try:
+        panels = Panel.objects.filter(panel_name=panel_name).order_by('-panel_version')
+        # highest panel version is the previous panel
```
Can you add to comment that this is because the newest panel isn't in the database yet? I was confused initially until I went through the code

Istaisa replied:
done

erikwaskiewicz commented on swgs/models.py:
```diff
@@ -371,4 +390,174 @@ class SomaticVEPAnnotations(AbstractVEPAnnotations):
+    def display_in_tier_two(self):
+        variant_gene = self.vep_annotations.first().transcript.gene
+        associated_panels = variant_gene.panels.all()
+        # if any of the associated panels are in a tier 1 panel, display
```
This comment is the same in all the filtering functions because of copy pasting (germline and somatic)

Istaisa replied:
fixed

erikwaskiewicz reviewed (changes requested):
I've added some minor comments in line, also need to add requests to the conda env

I couldnt get it to work locally but I think it's an issue with my panel setup, I'm nearly out of time though so I'm submitting this before finishing fully testing the GUI

erikwaskiewicz commented:
Got it to work, I needed to link the panels to my indication. Can you update the fixtures and give a quick install guide in the readme to cover it? Doesnt have to be comprehensive for now

I like the MNV feature :)

Istaisa commented:
- Addressed inline comments
- Added requests to env - builds from scratch locally but might be safest to pip install in the existing environment on Viglen
- Added some initial documentation 

What specifically do you mean by update the fixtures?

Istaisa commented:
Also just to note - I am going to do some long term versioning on the indications so that previous analyses maintain their original panel versions, but it's not a necessary to get this data for ruth

erikwaskiewicz reviewed (approved):
Looks good, happy with updates. For the fixtures I think I meant with the indications set up, but the instructions cover that
